### PR TITLE
raising an error if description is missing

### LINKED ISSUE #14: If %description section is left out a traceback is shown. Ideally, an error should tell you it is a required section.

### PR DESCRIPTION
**Pull request by mostafahussein**
This should fix #14. I was thinking about using `try() and catch()` instead of using `raise()`. what do you think ?

### CODE REVIEW

**Comment by openx-luis** [comment]
Looks good to me!

I do agree that the exception traceback looks ugly. Did you want to implement your idea of using try-catch to have more user-friendly error messages?

I would do it in `main`, maybe with something like this:
```
def main():
    brine = Brine()
    try:
        brine.load()
    except Exception as e:
        sys.stderr.write("Error: {0}\n".format(e.message))
        sys.exit(1)
    brine.save_files()
```

What do you think? Feel free to update your PR with this code, or even make a new PR with better error reporting.

**Comment by openx-luis** [comment]
One more comment along these lines. To avoid masking unexpected Exceptions, while we're at it we should also add a base `BrineError` exception class, and raise those in the code instead, to be caught in `main`. That way, the known errors can be printed in a good format, and unexpected exceptions still show up with the full traceback.

**Comment by mostafahussein** [comment]
Hello, 
I have added the following class:
```python
class BrineError(Exception):
     pass
```
Then inside `Brine` class i have added a static method which will responsible for print the error message then exist instead of duplicating the code all over the place :D
```python
@staticmethod
def beauty_error(msg):
    try:
        raise BrineError(msg)
    except BrineError as e:
        sys.stderr.write("Error: {0}\n".format(e.message))
        sys.exit(1)
```
I made it a static because there is no need to pass `self` to this method as i don't need it inside. Now whenever you need to raise an error message just call the above method within `Brine` class like this 
```python
self.beauty_error('Zombies Ahead!')
```
And the result would be:
```
Error: Zombies Ahead!
```
Feel free to change the method name if you would like to :D

**Comment by openx-luis** [comment]
Ok, that's not what I meant at all. :)

With what you're doing you might as well forgo the exceptions and just print the error and exit directly, in that staticmethod.

Back to what I was saying. Consider the case of someone who is trying to use `brine.load()` in a script. All of a sudden the program exits! That's clearly undesirable behavior. We want to allow the user of the `Brine` instance to catch the exception and act in accordance to what they want to do.

So, to summarize,
* try-except clause only in `main()` – we print error message here
* `brine.load()` throws `BrineError` exception, and user of script decides what to do with it.

Any more questions?

**Comment by mostafahussein** [comment]
My bad! I thought you meant in your second comment to handling errors outside main.
Anyway i have updated the code now you can take a look.
What about "let the user decides" ? I think when there is an error he should fix those errors then calling brine once again.

**Comment by openx-luis** [comment]
Looks good now! You have like 5 commits, though. Did you want to squash your changes into just 1-2 commits? Other than that, I'd say your PR is ready to merge.

**Comment by mostafahussein** [comment]
I have squashed it but i got duplicate the commit message without notice, is it ok to merge it?
